tests/test_swift.py

```diff
@@ -87,35 +73,139 @@ def setUp(self):
         self.swift = swift.Swift(APP_INFOS)
 
     def test_init(self):
+        self.assertEqual(self.swift.appInfos, APP_INFOS)
         self.assertIsInstance(self.swift.mainWindow, QMainWindow)
         self.assertIsInstance(self.swift.centralWidget, QLabel)
         for name in APP_INFOS:
+            self.assertIn(name, self.swift._dockWidgets)
             self.assertIn(name, self.swift._apps)
         for channel in self.channels:
             self.assertIn(channel, self.swift._subscribers)
 
+    def test_app_names(self):
+        appNames = self.swift.appNames()
+        for name in APP_INFOS:
+            self.assertIn(name, appNames)
+
     def test_destroy_app(self):
         for name in APP_INFOS:
             self.swift.destroyApp(name)
             self.assertNotIn(name, self.swift._apps)
 
+    def test_update_frames(self):
+        self.swift._apps["app1"].frames.return_value = (QWidget(),)
+        self.swift.updateFrames("app1")
+
+    def test_channel_names(self):
+        channelNames = self.swift.channelNames()
+        for channelName in self.channels:
+            self.assertIn(channelName, channelNames)
+
+    def test_subscriber_names(self):
+        for channel in self.channels:
+            subscriberNames = self.swift.subscriberNames(channel)
+            for name, info in APP_INFOS.items():
+                if channel in info.channel:
+                    self.assertIn(name, subscriberNames)
+
+    def test_unsubcribe(self):
+        self.assertEqual(self.swift.unsubscribe("app1", "ch1"), True)
+        self.assertNotIn("app1", self.swift._subscribers["ch1"])
+        self.assertEqual(self.swift.unsubscribe("app2", "ch1"), False)
+
+    def test_swiftcall(self):
+        QMessageBox.warning = MagicMock(return_value=QMessageBox.Ok)
+        self.swift._swiftcall("app1", json.dumps({"call": "appNames", "args": {}}))
+        self.swift._swiftcall("app1", json.dumps({
+            "call": "createApp",
+            "args": {"name": "app3", "info": {"module": "module2", "cls": "cls2"}}
+        }))
+        self.swift.callWithSerializable = MagicMock()
+        self.swift.getSerializable = MagicMock(return_value=APP_INFOS["app1"])
+        self.swift._swiftcall("app1", json.dumps({"call": "getSerializable", "args": {}}))
+        self.swift._swiftcall("app1", json.dumps({"call": "_broadcast", "args": {}}))
+        QMessageBox.warning.return_value = QMessageBox.Cancel
+        self.swift._swiftcall("app1", json.dumps({"call": "appNames", "args": {}}))
```

Comment:
Hmm.. maybe I should implement the test codes for swiftcalls since I implemented it. Let's just leave this codes to keep the 100% coverage.
I will implemented the test codes after this PR.